src/main/io/vtx_smartaudio.c

```diff
@@ -888,11 +873,47 @@ static bool vtxSAIsReady(const vtxDevice_t *vtxDevice)
     return vtxDevice!=NULL && !(saDevice.version == 0);
 }
 
+static bool saValidateBandAndChannel(uint8_t band, uint8_t channel)
+{
+    return (band >= VTX_SMARTAUDIO_MIN_BAND && band <= vtxSmartAudio.capability.bandCount &&
+            channel >= VTX_SMARTAUDIO_MIN_CHANNEL && channel <= vtxSmartAudio.capability.channelCount);
+}
+
+uint8_t saGetFreqMode(){
+    if (saDevice.mode & SA_MODE_GET_FREQ_BY_FREQ) {
+        if (saDevice.vtxTableFreqModeOn) {
+            return SA_FREQ_MODE_VTXTABLE_FREQ;
+        } else {
+            return SA_FREQ_MODE_USER;
+        }
+    } else {
+        return SA_FREQ_MODE_VTXTABLE_INDEX;
+    }
+}
+
 static void vtxSASetBandAndChannel(vtxDevice_t *vtxDevice, uint8_t band, uint8_t channel)
 {
     UNUSED(vtxDevice);
     if (saValidateBandAndChannel(band, channel)) {
-        saSetBandAndChannel(band - 1, channel - 1);
+        if (saDevice.vtxTableFreqModeOn || !saDevice.vtxTableFrequenciesAreDefault){
+            saDevice.vtxTableFreqModeOn = true;
+            //device either currently in vtxtable freq mode, or in user freq mode but about to enter vtxtable freq mode
+            uint16_t freq = vtxCommonLookupFrequency(&vtxSmartAudio, band, channel);
+            if(saValidateFreq(freq)){
```

Comment:
That old thing again - please use spaces after `if` and on the outside of curly braces.